Scripts/RemediationScripts/Remediate-SetClusterProtectionLevelForServiceFabric.ps1

```diff
@@ -417,73 +510,143 @@ function Set-ClusterProtectionLeveltoEncryptandSignforServiceFabric
                     if($paramValue.Parameters.Value -eq 'EncryptAndSign')
                     {
                         $ServiceFabricRemediated += $ServiceFabric
+                        $logResource = @{}	
+                        $logResource.Add("ResourceGroupName",($_.ResourceGroupName))	
+                        $logResource.Add("ResourceName",($_.ResourceName))	
+                        $logRemediatedResources += $logResource	
                     }
                     else
                     {
                         $ServiceFabricSkipped += $ServiceFabric
+                        $logResource = @{}	
+                        $logResource.Add("ResourceGroupName",($_.ResourceGroupName))	
+                        $logResource.Add("ResourceName",($_.ResourceName))
+                        $logResource.Add("Reason", "Error setting cluster protection level to Encrypt and Sign: [$($ServiceFabric)]")               
+                        $logSkippedResources += $logResource	
+
                     }
                 }
             }
             catch
             {
                 $ServiceFabricSkipped += $ServiceFabric
-            }
-        }
+                $logResource = @{}	
+                $logResource.Add("ResourceGroupName",($_.ResourceGroupName))	
+                $logResource.Add("ResourceName",($_.ResourceName))	
+                $logResource.Add("Reason","Encountered error while setting cluster protection level to Encrypt and Sign Service Fabric")    	
+                $logSkippedResources += $logResource	
+                Write-Host "Skipping this resource..." -ForegroundColor $([Constants]::MessageType.Warning)	
+                }
+             }
 
         Write-Host $([Constants]::DoubleDashLine)
-        Write-Host "Remediation Summary:`n" -ForegroundColor $([Constants]::MessageType.Info)
         
-        if ($($ServiceFabricRemediated | Measure-Object).Count -gt 0)
+        
+        if($AutoRemediation)
         {
-            Write-Host "Successfully set the ClusterProtectionLevel to 'EncryptAndSign' on the following ServiceFabric(s) in the subscription:" -ForegroundColor $([Constants]::MessageType.Update)
-           
-            $ServiceFabricRemediated | Format-Table -Property $colsProperty -Wrap
+            if ($($ServiceFabricRemediated | Measure-Object).Count -gt 0)
+            {
+                $ServiceFabricRemediated | Format-Table -Property $colsProperty -Wrap
```

Comment:
remove this statement, we don't need to show the table when auto remediation switch is on